pandas/core/sparse/series.py

```diff
@@ -279,8 +280,18 @@ def __array_wrap__(self, result, context=None):
         else:
             fill_value = self.fill_value
 
+        # GH 14167
+        # Since we are returning a dense representation of
+        # SparseSeries sparse_index might not align when calling
+        # ufunc on the array. There doesn't seem to be a better way
+        # to do this unfortunately.
```

Comment:
This wasn't intuitive to me but the __array_*__ functions defined in numpy don't work quite as expected in certain circumstances. `np.abs(s)` is the one that was throwing me for a loop.

```
s = SparseArray([1,-2,2,3], fill_value=-2)
np.abs(s)
```

What would happen to the dense version of the array would be that it would have 3 npoints when really it should only have 2. By zeroing out the sparse_index it fixes the problem, because it relies on the dense index instead going around the problem.

Reply:
these should simply operate on the sp_values, not the on the densified (or maybe its possible that *some* classes of ufuncs could operate on the dense, but not sure)

Reply:
so the expected behavior is `np.abs(SparseArray([1,-2,2,3])) == array([1,2,3])` ?